File: Data Structures & Algorithms/remove-element/submission-2.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size();
        int r = n;
        int i = 0;
        for (i = 0; i < n; ++i) {
            if (nums[i] == val) {
                swap(nums[i], nums[n - 1]);
                r--;
                n--;
                --i;
            }
        }
        return r;
    }
};

File: Data Structures & Algorithms/majority-element-ii/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        m = {}
        ans = set()
        treshold = len(nums) // 3
        for n in nums:
            m[n] = 1 + m.get(n, 0)
            if m[n] > treshold:
                ans.add(n)
        return list(ans)